### PR TITLE
Revamped examples

### PR DESCRIPTION
**Pull request by okrause**
This PR revamps all examples heavily.

Goal is to showcase all the functionality, typical use cases and provide foundational examples for user cut/paste and customisation. It is based on typical workflows I observed by multiple users and stuff I use myself often. 

Instead of overwriting the existing examples, it creates a no examples/gcp.new folder and leaves examples/gcp intact. This is to allow review and discussion of proposed changes. Eventually, it is intended to replace the current examples.

Documentation of the individual examples can be found on the Readme.md file within examples/gcp.new and within the main resource file within each examples folder.

It also got some references to the upcoming Active Directory datasource, which was recently commited particially .

I tested all examples once or twice.

### CODE REVIEW

**Review by suhasbshekar — APPROVED** [review]

File: examples/gcp.new/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# CVS/GCP Terraform provider examples
+
+This folder contains examples for the resources (ActiveDirectory, Volumes, Snapshots) and
+the data sources (Volumes, ActiveDirectory(future)) the provider supports.
+
+Each example is self-contained and doen't require files from other folders (provider.tf
+might be an expection). The tf-file named after the resource of interest in the example
```
Comment (lonico) [inline examples/gcp.new/README.md]: exception

File: examples/gcp.new/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    ├── advanced - More complex example using variables and outputs
+    ├── datasource - Shows how to query an existing volume for later TF use
+    ├── minimal - Most simple volume example. Can be considered as "Hello World"
+    ├── volume-batch - Creates multiple volumes out of a CSV file
+    ├── volume-nfs - Creates an NFS volume
+    └── volume-smb - Creates an SMB volume
+```
+
+## Usage
+Change into respective folder and change provider.tf to suite your GCP project. Next look for the "locals" block in the *.tf files and customize it to fir your environment. Then run "terraform init" followed by "terraform apply".
```
Comment (lonico) [inline examples/gcp.new/README.md]: typo: fir --> for

File: examples/gcp.new/active_directory/minimal/active_directory.tf [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# This example create an Active Directory entry for the specified region
```
Comment (lonico) [inline examples/gcp.new/active_directory/minimal/active_directory.tf]: create --> creates

File: examples/gcp.new/active_directory/minimal/active_directory.tf [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# Please note, that CVS will not evaluate the correctness of this data
+# until the frist SMB volume using this entry will be created
```
Comment (lonico) [inline examples/gcp.new/active_directory/minimal/active_directory.tf]: first

File: examples/gcp.new/active_directory/minimal/active_directory.tf [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  provider = netapp-gcp
+  region = local.region
+  username = local.ad_username
+  password = local.ad_password
+  domain = local.ad_domain
+  dns_server = local.ad_dns_server
+  net_bios = local.ad_net_bios
+  organizational_unit = local.ad_organizational_unit
+  site = local.ad_site
+}
```
Comment (lonico) [inline examples/gcp.new/active_directory/minimal/active_directory.tf]: newline missing

File: examples/gcp.new/snapshot/snapshot.tf [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# This example creates an snapshot of a volume specified via a volume data source
```
Comment (lonico) [inline examples/gcp.new/snapshot/snapshot.tf]: an --> a

File: examples/gcp.new/volume/advanced/variables.tf [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  type        = string
+  description = "Organizational Unit."
+  default     = "CN=Computers"
+}
+
+variable "ad_site" {
+  type        = string
+  description = "Active Directory Site"
+  default     = ""
+}
```
Comment (lonico) [inline examples/gcp.new/volume/advanced/variables.tf]: newline

File: examples/gcp.new/volume/datasource/datasource.tf [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+output "protocol_type"  {
+  value       = data.netapp-gcp_volume.myvolume.mount_points[0].protocol_type
+  description = "The protocol type of the export."
+}
+
+# If volume got multiple mount points (e.g. NFSv3 and NFSv4), use below method to get data for specific protocol
+output "exportfull" {
+  value       = [for x in data.netapp-gcp_volume.myvolume.mount_points: "${x.server}:${x.export}" if x.protocol_type == "NFSv3"][0]
+  description = "The full export path of the cloud volume."
+}
```
Comment (lonico) [inline examples/gcp.new/volume/datasource/datasource.tf]: newline
(If you're using vscode, there is a setting to always insert one)

File: examples/gcp.new/volume/volume-smb/volume-smb.tf [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+resource "netapp-gcp_volume" "gcp-volume-smb" {
+  name = local.volume_name
+  region = local.region
+  protocol_types = ["SMB"]
+  network = local.network
+  size = local.size
+  service_level = local.service_level
+
+  # Advise: Since SMB volumes can only be created if an Active Directory connection exists for the region,
```
Comment (lonico) [inline examples/gcp.new/volume/volume-smb/volume-smb.tf]: Advice

**Review by lonico — CHANGES_REQUESTED** [review]
looking great.  Some nitpicks

**Comment by okrause** [comment]
Fixed the typos mentioned above. Many thanks for the VScode "newline at EOF" hint. Very appreciated.

**Pull request merged**